src/main/scala/com/databricks/spark/xml/util/TypeCast.scala

```diff
@@ -119,7 +119,15 @@ private[xml] object TypeCast {
       map(supportedXmlTimestampFormatters :+ _).getOrElse(supportedXmlTimestampFormatters)
     formatters.foreach { format =>
       try {
-        return Timestamp.from(ZonedDateTime.parse(value, format).toInstant)
+        // If format is not in supported and no timezone in format, use default Spark timezone
+        if (!supportedXmlTimestampFormatters.contains(format) && Option(format.getZone).isEmpty) {
+          return Timestamp.from(
+            ZonedDateTime.parse(value, format.withZone(ZoneId.of(options.timezone.get))).toInstant
```

Comment:
Need to deal with timezone not being specified - I suppose throw an error explicitly?

Reply:
With the current approach to put the timezone in `XmlRelation` of  `createRelation()`, isn't it better the create a default timezone in case `spark.sql.session.timeZone` is not set?

Reply:
Hm, maybe, I just feel like that's an error. You are parsing times that are ambiguous without a timzeone, and didn't give a timezone - just assuming "UTC" or something doesn't quite seem right vs highlighting the error.